### PR TITLE
Add unit test cases for scheduler/algorithm/predicates.

### PR DESCRIPTION
**Pull request by tizhou86**
**What this PR does / why we need it**:
Add unit test cases for scheduler/algorithm/predicates for more code coverage.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 1]
```diff
@@ -3470,6 +3470,80 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			},
+		},
+	}
+
+	// specify a node with pressure condition on
+	pressureNode := &v1.Node{
+		Status: v1.NodeStatus{
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   "PIDPressure",
```
Comment (k82cn) [inline pkg/scheduler/algorithm/predicates/predicates_test.go]: replace with const

File: pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 2]
```diff
@@ -3470,6 +3470,80 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+			name:     "pod not schedulable on node with pressure condition on",
+		},
+	}
+	expectedFailureReasons := []algorithm.PredicateFailureReason{ErrNodeUnderPIDPressure}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fits, reasons, err := CheckNodePIDPressurePredicate(test.pod, PredicateMetadata(test.pod, nil), test.nodeInfo)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
```
[the remark this answers is not in the document]
Comment (tizhou86) [reply pkg/scheduler/algorithm/predicates/predicates_test.go]: Sure, thanks for the advice!

File: pkg/scheduler/algorithm/predicates/predicates_test.go (hunk 1 above)
Comment (tizhou86) [reply pkg/scheduler/algorithm/predicates/predicates_test.go]: OK

File: pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 3]
```diff
@@ -3470,6 +3470,80 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+		},
+	}
+
+	// specify a node with no pid pressure condition on
+	noPressureNode := &v1.Node{
+		Status: v1.NodeStatus{
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   v1.NodeReady,
+					Status: "True",
```
Comment (k82cn) [inline pkg/scheduler/algorithm/predicates/predicates_test.go]: we also have const for status :)

Comment (tizhou86) [reply pkg/scheduler/algorithm/predicates/predicates_test.go]: Sure, I'll fix it.

File: pkg/scheduler/algorithm/predicates/predicates_test.go [hunk 4]
```diff
@@ -3470,6 +3470,80 @@ func TestPodSchedulesOnNodeWithDiskPressureCondition(t *testing.T) {
...
+	}
+
+	tests := []struct {
+		pod      *v1.Pod
+		nodeInfo *schedulercache.NodeInfo
+		fits     bool
+		name     string
+	}{
+		{
+			pod:      pod,
```
Comment (resouer) [inline pkg/scheduler/algorithm/predicates/predicates_test.go]: Seems you only defined one pod for this test. Then it does not need to be part of test case.

Comment (tizhou86) [reply pkg/scheduler/algorithm/predicates/predicates_test.go]: Revised, thanks!

**Pull request merged**